table/filter_block.h

```diff
@@ -123,6 +123,29 @@ class FilterBlockReader {
   virtual void CacheDependencies(bool /*pin*/,
                                  const SliceTransform* /*prefix_extractor*/) {}
 
+  virtual bool IsFilterCompatible(
+      const ReadOptions& /*read_options*/, const Slice& /*user_key*/,
+      const Comparator* /*comparator*/) {
+    return true;
+  }
+
+  bool RangeMayExist(const ReadOptions& read_options, const Slice& user_key,
```

Comment:
It will be nice if filter classes don't need to depend on ReadOptions. It seems that all what we need here is upper bound. Then we can use `const Slice* upper_bound` instead.